File: api-tester/src/customEntitlementComputation/kotlin/com/revenuecat/apitester/java/PurchasesAPI.java
package com.revenuecat.apitester.java;

import android.content.Context;

import com.revenuecat.purchases.Purchases;
import com.revenuecat.purchases.PurchasesConfigurationForCustomEntitlementsComputationMode;

import java.util.concurrent.ExecutorService;

@SuppressWarnings({"unused"})
final class PurchasesAPI {

    static void checkConfiguration(final Context context,
                                   final ExecutorService executorService) {
        Purchases.configureInCustomEntitlementsComputationMode(context, "", "");
        Purchases.configureInCustomEntitlementsComputationMode(
                new PurchasesConfigurationForCustomEntitlementsComputationMode.Builder(
                        context,
                        "",
                        ""
                ).showInAppMessagesAutomatically(false)
                        .pendingTransactionsForPrepaidPlansEnabled(false)
                        .build()
        );
    }

    static void check(final Purchases purchases) {
        purchases.switchUser("newUserID");
    }
}

File: api-tester/src/customEntitlementComputation/kotlin/com/revenuecat/apitester/kotlin/PurchasesAPI.kt
package com.revenuecat.apitester.kotlin

import android.content.Context
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.PurchasesConfigurationForCustomEntitlementsComputationMode

@Suppress("unused", "UNUSED_VARIABLE", "EmptyFunctionBlock")
private class PurchasesAPI {

    @Suppress("ForbiddenComment")
    @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
    fun checkConfiguration(context: Context) {
        val configured: Boolean = Purchases.isConfigured

        Purchases.configureInCustomEntitlementsComputationMode(context, apiKey = "", appUserID = "")
        Purchases.configureInCustomEntitlementsComputationMode(
            PurchasesConfigurationForCustomEntitlementsComputationMode
                .Builder(
                    context = context,
                    apiKey = "",
                    appUserID = "",
                )
                .applyObfuscatedAccountIdToSubscriptionChanges(true)
                .showInAppMessagesAutomatically(false)
                .pendingTransactionsForPrepaidPlansEnabled(false)
                .build(),
        )
    }

    fun check(purchases: Purchases) {
        purchases.switchUser("newUserID")
    }
}
